Source/GoBot/FeedStrategy/FeedStrategy.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  RecentStrategy.swift
+//  Planetary
+//
+//  Created by Martin Dutra on 12/4/22.
+//  Copyright © 2022 Verse Communications Inc. All rights reserved.
+//
+
+import Foundation
+import SQLite
+
+/// Defines a family of functions needed for fetching a feed of keyvalues from the database, i.e., the Home Feed
+///
+/// This strategy supports having a family of algorithms that implement fetching
+/// a feed in different and unique ways (one could fetch just posts, another one
+/// could fetch posts and follows, and so on), and making them interchangeable.
+protocol FeedStrategy {
+    /// Returns the total number of items in the feed
+    /// - parameter connection: the database connection needed to run queries in the database
+    /// - parameter userId: the ID of the user for which the feed will be calculated from
+    ///
+    /// It must be the same number of items that fetchKeyValues and fetchKeys returns.
+    func countNumberOfKeys(connection: Connection, userId: Int64) throws -> Int
```

Comment:
Maybe this could just be called `numberOfKeys(`?

Reply:
I don't know about this one. According to the Swift [API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage), we should "name functions and methods according to their side-effects, those without side-effects should read as noun phrases [and] those with side-effects should read as imperative verb phrases".

But it is not clear how should a side-effect be defined. It can be something that changes the state of the app (I think we all agree about this), but what about this kind of functions that make expensive operations in the database, do they have "side-effects".

I left it countNumberOfKeys because I worry that just numberOfKeys can be read as a simple function that can be called without worrying to much. **I'm open to discuss this**.

Reply:
Just one more thought that popped into my head: I think I have used verbs like `calculate` or `compute` in the past to try to communicate expensive operations. So like `calculateNumberOfKeys`. Not sure if it's that much clearer though 🤷‍♂️